lxd/backup.go

```diff
@@ -0,0 +1,297 @@
+package main
+
+import (
+	"archive/tar"
+	"bytes"
+	"io"
+	"os"
+	"path/filepath"
+	"time"
+
+	yaml "gopkg.in/yaml.v2"
+
+	"github.com/lxc/lxd/lxd/db"
+	"github.com/lxc/lxd/lxd/state"
+	"github.com/lxc/lxd/shared"
+	"github.com/lxc/lxd/shared/api"
+	"github.com/pkg/errors"
+)
+
+var ErrMissingIndexFile = errors.New("missing index.yaml")
```

Comment:
Also, this is used once in the entire codebase, so I'd just use a good old `return fmt.Errorf()` instead of wasting memory by keeping this as a global variable.